### PR TITLE
Requests, Events: Add support for Audio Active

### PR DESCRIPTION
**Pull request by WizardCM**
This is used to determine whether a Browser source has "Control audio via OBS" enabled. If true, the source should appear in a mixer.

### CODE REVIEW

**Comment by tt2468** [comment]
Would you be able to convert all request and event properties to camelCase?

For source names, we have been using `sourceName`. `audio_enabled` can be `audioEnabled`, and in `GetAudioActive`, the request does not have to return the source name, since the caller already has it in order to make the request.

**Comment by WizardCM** [comment]
Sure, I basically copy-pasted the mute code, I can adapt as needed.

File: src/WSRequestHandler_Sources.cpp [hunk 1]
```diff
@@ -341,6 +341,41 @@ RpcResponse WSRequestHandler::ToggleMute(const RpcRequest& request)
 	return request.success();
 }
 
+/**
+* Get the audio's active status of a specified source.
+*
+* @param {String} `source` Source name.
```
Comment (tt2468) [inline src/WSRequestHandler_Sources.cpp:347]:
```suggestion
* @param {String} `sourceName` Source name.
```

File: src/WSRequestHandler_Sources.cpp [hunk 2]
```diff
@@ -341,6 +341,41 @@ RpcResponse WSRequestHandler::ToggleMute(const RpcRequest& request)
...
+* @return {boolean} `audioActive` Audio active status of the source.
+*
+* @api requests
+* @name GetAudioActive
+* @category sources
+* @since 4.9.0
+*/
+RpcResponse WSRequestHandler::GetAudioActive(const RpcRequest& request)
+{
+	if (!request.hasField("source")) {
```
Comment (tt2468) [inline src/WSRequestHandler_Sources.cpp:359]:
```suggestion
	if (!request.hasField("sourceName")) {
```

File: src/WSRequestHandler_Sources.cpp [hunk 3]
```diff
@@ -341,6 +341,41 @@ RpcResponse WSRequestHandler::ToggleMute(const RpcRequest& request)
 	return request.success();
 }
 
+/**
+* Get the audio's active status of a specified source.
+*
+* @param {String} `source` Source name.
+*
+* @return {String} `name` Source name.
```
Comment (tt2468) [inline src/WSRequestHandler_Sources.cpp:349]:
```suggestion
```

File: src/WSRequestHandler_Sources.cpp [hunk 4]
```diff
@@ -341,6 +341,41 @@ RpcResponse WSRequestHandler::ToggleMute(const RpcRequest& request)
...
+* @category sources
+* @since 4.9.0
+*/
+RpcResponse WSRequestHandler::GetAudioActive(const RpcRequest& request)
+{
+	if (!request.hasField("source")) {
+		return request.failed("missing request parameters");
+	}
+
+	QString sourceName = obs_data_get_string(request.parameters(), "source");
```
Comment (tt2468) [inline src/WSRequestHandler_Sources.cpp:363]:
```suggestion
	QString sourceName = obs_data_get_string(request.parameters(), "sourceName");
```

**Comment by Palakis** [comment]
Looks good to me, haven't tested yet but i'll trust @tt2468's review and testing

File: src/WSEvents.cpp [hunk 5]
```diff
@@ -124,7 +124,6 @@ void WSEvents::FrontendEventHandler(enum obs_frontend_event event, void* private
 		case OBS_FRONTEND_EVENT_FINISHED_LOADING:
 			owner->hookTransitionPlaybackEvents();
 			break;
-
```
Comment (WizardCM) [inline src/WSEvents.cpp:127]: @tt2468 I hope you're not keeping this :P

**Pull request merged**